src/node_crypto_bio.h

```diff
@@ -96,6 +96,20 @@ class NodeBIO : public MemoryRetainer {
     return length_;
   }
 
+  // Provide a hint about the size of the next pending set of writes. TLS
+  // writes records of a maximum length of 16k of data plus a 5-byte header,
+  // a MAC (up to 20 bytes for SSLv3, TLS 1.0, TLS 1.1, and up to 32 bytes
+  // for TLS 1.2), and padding if a block cipher is used.  If there is a
+  // large write this will result in potentially many buffers being
+  // allocated and gc'ed which can cause long pauses.   By providing a
+  // guess about the amount of buffer space that will be needed in the
+  // next allocation this overhead is removed.
+  inline void set_allocate_hint(size_t size) {
+    if (size >= 16 * 1024) {
+      allocate_hint_ = ((size / (1024 * 16) + 1) * ((1026*16) + 5 + 32));
```

Comment:
Can we maybe move this computation (+ the related parts of the comment) into a helper function in `tls_wrap.cc`, since it’s specific to TLS?

Reply:
Thanks for getting back so fast. The BIO is only used by OpenSSL so I don't this is going to be used by anything else.

Reply:
Yeah, but it’s also used for things that are not related to TLS (e.g. in `node_crypto.cc`) – it feels odd to put TLS-specific computations into it in that case.

If you feel that moving the computation into the TLS file is unnecessary (which is understandable), I’d suggest referring to TLS in the method name somehow to indicate that this is not a “generic” allocation hint.

Reply:
I've changed it to set_allocate_tls_hint